homeassistant/components/huum/climate.py

```diff
@@ -0,0 +1,131 @@
+
+    async_add_entities([HuumDevice(entry_status, _huum_handler)])
+
+
+class HuumDevice(ClimateEntity):
+    """Representation of a heater."""
+
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_max_temp = 110
+    _attr_min_temp = 40
+
+    _target_temperature = 40
+
+    def __init__(self, status: HuumStatusResponse, _huum_handler: Huum) -> None:
+        """Initialize the heater."""
+        self._status = status
+        self._huum_handler = _huum_handler
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device, if any."""
+        return "Huum Sauna"
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if self._status.status == SaunaStatus.ONLINE_HEATING:
+            return HVACMode.HEAT
+        return HVACMode.OFF
+
+    @property
+    def icon(self) -> str:
+        """Return nice icon for heater."""
+        if self.hvac_mode == HVACMode.HEAT:
+            return "mdi:radiator"
+        return "mdi:radiator-off"
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        return self._status.temperature
+
+    @property
+    def target_temperature(self) -> int:
+        """Return the temperature we try to reach."""
+
+        return self._status.target_temperature or self._target_temperature
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set hvac mode."""
+        if hvac_mode == HVACMode.HEAT:
+            temperature = max(self.min_temp, self.target_temperature)
+            await self._turn_on(temperature)
+        elif hvac_mode == HVACMode.OFF:
+            await self._huum_handler.turn_off()
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        self._target_temperature = temperature
+
+        if self.hvac_mode == HVACMode.HEAT:
+            await self._turn_on(temperature)
+
+    async def async_update(self) -> None:
+        """Get the latest status data.
+
+        We get the latest status first from the status endpoints of the sauna.
+        If that data does not include the temperature, that means that the sauna
+        is off, we then call the off command which will in turn return the temperature.
+        This is a workaround for getting the temperature as the Huum API does not
+        return the target temperature of a sauna that is off, even if it can have
+        a target temperature at that time.
+        """
+        self._status = await self._huum_handler.status_from_status_or_stop()
+
+    async def _turn_on(self, temperature) -> None:
```

Comment:
Isn't the function for turning off missing?

Reply:
Ah, actually, now that I look at it again, the turn_off has some extra handling due to also handling the case of an open door, that is why turn_on has extra handling here and not turn_off, since that does not care about the door.

Reply:
I misread this as the default `turn_on` service which you are not implementing I see now (which I think you should implement)

Reply:
Would it make sense to implement a turn_on feature as `heat` is already implemented in this case? The sauna itself is always on, but either heating or not heating. Already the `off` implementation here is borderline wrong, but there is no better option, since `cool` is not really correct either.